### ISSUE TITLE
Component RadioControl stop to working in WP 5.8.1

### ISSUE THREAD

**Issue by rubengc** [WordPress/gutenberg#35107]
Hi there!

From latest update, the RadioControl component stop to working and its value never gets updated.

I'm using the "selected" attribute which was working completely fine until this update.

Is there any change made in 5.8.1 to this component?

I checked the WP docs and seems the attributes still the same as I was using from the release of Gutenberg so I do not know exactly what's happening...

**Comment by Mamaduka** [comment]
Hi, @rubengc

Can you share an example code for testing?

**Comment by rubengc** [comment]
Hi @Mamaduka 

Of course, I found it in my plugin GamiPress, every control (checkboxes, even a multicheckboxes we created) is working fine except the RadioControl

Here is a test website with GamiPress installed where you can test it in any post editor:
https://tastewp.com/new/?pre-installed-plugin-slug=gamipress

The steps to reproduce are:
1 - Create a post
2 - Add the block Single Achievement
3 - Try to change the "Layout" or "Alignment" option (both are RadioControl)

The thing is that I have not updated this code in months, and was working completely fine until 5.8.1:
https://github.com/rubengc/GamiPress-Gutenberg-Blocks/blob/master/src/block/block.js#L278-L284

Hope we can find a way to get it fixed :)

**Comment by rubengc** [comment]
Hi @Mamaduka any news about this?

This issue seems extremely strange and I just did a test to rollback to WordPress 5.7.9 where this control works perfectly, seems that issue was introduced in version 5.8 and above

**Comment by rubengc** [comment]
I added more info here: https://github.com/WordPress/gutenberg/issues/35107#issuecomment-926528785

**Comment by rubengc** [comment]
Can anyone review this?

**Comment by t-hamano** [comment]
Hi @rubengc, 

I have carefully followed your code and the changes made in WordPress 5.8 regarding `RadioControl`.

First, the versions of Gutenberg in WordPress 5.7 and 5.8 are as follows:

- WordPress 5.7: Gutenberg 9.9
- WordPress 5.8: Gutenberg 10.7

And the following PR was merged in 10.0, between Gutenberg 9.9 and Gutenberg 10.7:
[Components: Allow extra props for RadioControl component #28631](https://github.com/WordPress/gutenberg/pull/28631)

I further checked the code you reported and confirmed that you have already solved the problem by not passing the value prop to RadioControl.

https://github.com/rubengc/GamiPress-Gutenberg-Blocks/commit/27aaa75ea1836861e1974492ba87c25094d535e9#diff-8d293b10b538e055f46f2c756bd7615a35ae7b33882a80fcd0eb96588c08df25R217-R221

Your commit is a change to not pass the value prop in the case of the RadioControl component.
In other words, in WordPress 5.7, `RadioControl` doesn't accept additional props, so passing a value prop did not cause any problems.
But in WordPress 5.8, additional props are allowed, so `RadioControl` now receives value props that it is not supposed to.

In this light, it makes sense that your commit fixed the problem.

As stated above, I would like to close this issue as I consider this to be a specification change rather than a bug on the component side.

If you have any questions, please feel free to reopen.
